### PR TITLE
VAGOV-2025: Side Nav Level 3 & Deeper Fix

### PR DESCRIPTION
**Pull request by acabouet**
## Description
Makes main side nav items 3 levels down and deeper work the way they do on prod. 

## Screenshots
Screen Shot 2019-03-22 at 5 11 41 PM

## Acceptance criteria
- [ ] See ticket: https://va-gov.atlassian.net/browse/VAGOV-2025

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by ethanteague — APPROVED** [review]

File: src/site/navigation/sidebar_nav.drupal.liquid [hunk 1]
```diff
@@ -56,7 +63,38 @@
...
+                                    <li>
+                                        <a {% if entityUrl.path == link.url.path %} class="usa-current" {% endif %}
+                                                href="{{ link.url.path }}" onClick="recordEvent({ event: 'nav-sidenav' });">{{ link.label }}</a>
+                                    </li>
+                                {% endfor %}
+                            </ul>
+                    </li>
+                        {% else %}
+                    </li>
+                        {% endif %}
```
Comment (beeyayjay) [inline src/site/navigation/sidebar_nav.drupal.liquid]: You could put the `</li>` after the endif and get rid of the else.

**Review by beeyayjay — APPROVED** [review]
This looks great! I stepped down into the deepest links and clicked around randomly, and on every page, the menu looked and responded exactly like the live site. The code looks really solid, too.

File: src/site/filters/liquid.js [hunk 2]
```diff
@@ -68,4 +68,68 @@ module.exports = function registerFilters() {
     const id = `vha_${facilityLocatorApiId}`;
     return JSON.stringify(id);
   };
+
+  // Find the current path in an array of nested link arrays and then return it's depth + it's parent and children
+  liquid.filters.findCurrentPathDepth = (linksArray, currentPath) => {
+    const getDeepLinks = (path, linkArr) => {
```
Comment (jbalboni) [inline src/site/filters/liquid.js]: Did you have any luck with Aurora's recursive approach she posted in Slack? I think that's the right approach for something like this, but I understand if we're crunched for time.

File: src/site/filters/liquid.js [hunk 3]
```diff
@@ -68,4 +68,68 @@ module.exports = function registerFilters() {
...
+          for (let b = 0; b < linkArr[a].links.length; b += 1) {
+            if (linkArr[a].links[b].url.path === path) {
+              deepObj.depth = 2;
+              return deepObj;
+            }
+            if (linkArr[a].links[b].links) {
+              for (let c = 0; c < linkArr[a].links[b].links.length; c += 1) {
+                if (linkArr[a].links[b].links[c].url.path === path) {
+                  deepObj.depth = 3;
+                  deepObj.links = linkArr[a].links[b];
```
Comment (jbalboni) [inline src/site/filters/liquid.js]: Seems like parent might be a better name than links?

File: src/site/filters/liquid.js [hunk 4]
```diff
@@ -68,4 +68,68 @@ module.exports = function registerFilters() {
...
+                }
+              }
+            }
+          }
+        }
+      }
+      return false;
+    };
+
+    return JSON.stringify(getDeepLinks(currentPath, linksArray));
```
Comment (jbalboni) [inline src/site/filters/liquid.js]: Does liquid force filters to return strings? Or could you return the object here?

**Review by jbalboni — COMMENTED** [review]
Couple questions, but at first pass at this looks good. I can test it locally tomorrow, too.

File: src/site/filters/liquid.js (hunk 2 above)
Comment (acabouet) [reply src/site/filters/liquid.js]: I gotta re-work it a bit to get the data structure I need but I think this needs to be merged in ASAP if it's working. And then I can push up a refactor post-launch if that works.

**Comment by jbalboni** [comment]
@acabouet The button to show the sidebar menu on mobile is missing now. Can you take a look at that?

**Review by jbalboni — APPROVED** [review]
This is fine temporarily, but we definitely need to update `findCurrentPathDepth` to be more concise long term.

**Pull request merged**